rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -284,6 +299,46 @@ impl Iterator for CanisterOutputQueuesIterator<'_> {
     }
 }
 
+/// Kinds of canister inputs returned by `CanisterQueues::pop_input()` /
+/// `CanisterQueues::peek_input()`: in addition to the regular ingress messages
+/// and canister requests / responses, `pop_input()` / `peek_input()` may also
+/// return concise "reject response for callback ID" messages.
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum CanisterInput {
+    Ingress(Arc<Ingress>),
+    Request(Arc<Request>),
+    Response(Arc<Response>),
+    /// A concise reject response meaning "call deadine has expired".
+    #[allow(dead_code)]
+    DeadlineExpired(CallbackId),
```

Comment:
Do we intend to start using this in a follow up PR?

Reply:
Yeah, I intend to implement callback deadline expiration right after this, since it can rely the exact mechanisms we're building for `ResponseDropped` in this PR.